### PR TITLE
Revert to StatusBarItem for select configuration

### PR DESCRIPTION
**Pull request by browntarik**
Addresses: https://github.com/microsoft/vscode-cpptools/issues/10755

### CODE REVIEW

**Review by sean-mcmanus — DISMISSED** [review]

File: Extension/src/LanguageServer/ui_new.ts [hunk 1]
```diff
@@ -438,14 +436,27 @@ export class NewUI implements UI {
...
                 isCppPropertiesJson = activeEditor.document.fileName.endsWith("c_cpp_properties.json");
                 if (isCppPropertiesJson) {
                     vscode.languages.setTextDocumentLanguage(activeEditor.document, "jsonc");
                 }
             }
+
+            // It's sometimes desirable to see the config and icons when making changes to files with C/C++-related content.
+            // TODO: Check some "AlwaysShow" setting here.
+            this.ShowConfiguration = isCpp || isCppPropertiesJson ||
+                activeEditor.document.uri.scheme === "output" ||
```
Comment (Colengms) [inline Extension/src/LanguageServer/ui_new.ts:455]: Should this check if the output channel is one of ours?

Comment (Colengms) [reply Extension/src/LanguageServer/ui_new.ts:455]: Do we need to display something when these are open?  The reasoning isn't obvious to me.

**Review by Colengms — CHANGES_REQUESTED** [review]

File: Extension/src/LanguageServer/ui_new.ts (hunk 1 above)
Comment (browntarik) [reply Extension/src/LanguageServer/ui_new.ts:455]: I am not sure; I am unfamiliar with this code. @sean-mcmanus @bobbrow What do you guys think?

Comment (sean-mcmanus) [reply Extension/src/LanguageServer/ui_new.ts:455]: This is "pre-existing" code. I don' t know if there's a way to detect if the selected document output channel is one of ours or not. It seems low priority and shouldn't block this PR though. We could file an issue to track that if you want.

Comment (sean-mcmanus) [reply Extension/src/LanguageServer/ui_new.ts:455]: We also have an active issue to "always" show the configuration with some setting regardless of the active file.

Comment (bobbrow) [reply Extension/src/LanguageServer/ui_new.ts:455]: > I am not sure; I am unfamiliar with this code. @sean-mcmanus @bobbrow What do you guys think?

If this looks the same as what's currently in `ui.ts`, then I'm fine with what's here.  The goal of this change is to revert the configuration selection button back to its old place.

Comment (sean-mcmanus) [reply Extension/src/LanguageServer/ui_new.ts:455]: yeah

**Review by Colengms — APPROVED** [review]

**Review by sean-mcmanus — APPROVED** [review]

**Pull request merged**